lib/DataSet.js

```diff
@@ -668,25 +668,26 @@ DataSet.prototype._sort = function (items, order) {
  */
 DataSet.prototype.remove = function (id, senderId) {
   var removedIds = [],
-      i, len, removedId;
-
-  if (Array.isArray(id)) {
-    for (i = 0, len = id.length; i < len; i++) {
-      removedId = this._remove(id[i]);
-      if (removedId != null) {
-        removedIds.push(removedId);
+      removedItems = [],
+      ids = [],
+      i, len, itemId, item;
+
+  // force everything to be an array for simplicity
+  ids = Array.isArray(id) ? id : [id];
+
+  for (i = 0, len = ids.length; i < len; i++) {
+    item = this._remove(ids[i]);
+    if (item) {
+      itemId = item[this._fieldId];
+      if (itemId) {
```

Comment:
This brings in some old bugs. Like I want to remove {id: 0, from: 4, to: 5} from DataSet edges.